app/src/main/java/br/org/cesar/knot_setup_app/view/scan/ScanPresenter.java

```diff
@@ -57,35 +64,40 @@ public void onScanComplete(android.bluetooth.le.ScanResult result) {
                     }
                     deviceList.add(new BluetoothDevice(result.getDevice(),result.getRssi()));
                     Collections.sort(deviceList);
-                    mViewModel.callbackOnDeviceFound(deviceList);
+                    viewModel.callbackOnDeviceFound(deviceList);
                 }
 
                 @Override
                 public void onScanFail() {
```

Comment:
Is this method necessary? This method is empty and as I can see your class do not implement the interface it was declared.

Reply:
Not sure if I understood your statement, the class ScanFragment.java does implement the ViewModel interface that declares this method. Nevertheless, this method is used to update the list that displays the nearby Bluetooth devices (Things and Gateways).